Source/URLDecomposer.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  URLDecomposer.swift
+//  Kakapo
+//
+//  Created by Joan Romano on 31/03/16.
+//  Copyright © 2016 devlucky. All rights reserved.
+//
+
+import Foundation
+
+/**
+ A tuple holding components and query parameters, check `decomposeURL` for more details
+ */
+public typealias URLInfo = (components: [String : String], queryParameters: [NSURLQueryItem])
+
+/**
+ Checks and parses if a given `handlerPath` representation matches a `requestURLComponents`. Examples:
+ 
+ `/users/:id` with `/users/1` produces `components: ["id" : "1"]`
+ `/users/:id/comments` with `/users/1/comments` produces `components: ["id" : "1"]`
+ `/users/:id/comments/:comment_id` with `/users/1/comments/2?page=2&author=hector` produces `components: ["id": "1", "comment_id": "2"]` and `queryParameters: ["page": "2", "author": "hector"]`
+ 
+ - parameter handlerPath: the URL handler with dynamic paths
+ - parameter requestURLComponents: the components of the actual URL
+ 
+ - returns: a URL info object containing `components` and `queryParamaters`
+ */
+func decomposeURL(handlerPath: String, requestURLComponents: NSURLComponents) -> URLInfo? {
+    var components: [String : String] = [:]
```

Comment:
a bit difficult to understand, would you mind adding comments with the expected result after the expressions? like `let handlerSplittedPaths.... // ["/user/", ":id/comment", "commentId"]`